diff --git a/sudoers b/sudoers
index 0d7760b..292f63c 100644
--- a/sudoers
+++ b/sudoers
@@ -19,7 +19,10 @@
 ##
 ## Groups of users.  These may consist of user names, uids, Unix groups,
 ## or netgroups.
+##
+## Change this to your likings...
 # User_Alias	ADMINS = millert, dowdy, mikef
+User_Alias	ADMINS = %root
 
 ##
 ## Cmnd alias specification
@@ -62,6 +65,42 @@
 # Defaults!/usr/local/bin/sudoreplay !log_output
 # Defaults!/sbin/reboot !log_output
 
+# always prompt for a password
+Defaults timestamp_timeout = 0
+
+# The number of tries a user gets to enter his/her password before sudo logs the failure and exits
+Defaults passwd_tries = 1
+
+# always require password when a user runs sudo -l
+Defaults listpw = always
+
+# prevent a dynamically-linked executable from running further commands itself
+Defaults noexec
+
+# sudo will only run when the user is logged in to a real tty
+Defaults requiretty
+
+# don't allow root to run sudo
+# sudo displays a message: "sudo: sudoers specifies that root is not allowed to sudo"
+Defaults !root_sudo
+
+# prompt for the root password instead of the password of the invoking user
+Defaults rootpw
+
+Defaults secure_path="/usr/sbin:/sbin:/usr/bin:/bin"
+
+# ignore '.' or '' (current dir) in the PATH environment variable
+Defaults ignore_dot
+
+# send an alert on different occasions
+Defaults mailto=root, mail_badpass, mail_no_host, mail_no_perms, mail_no_user
+
+# additional logging
+Defaults logfile=/var/log/sudo.log, log_host, log_year
+
+Defaults editor=/usr/bin/vim
+Defaults passwd_timeout=1
+
 ##
 ## Runas alias specification
 ##
@@ -69,7 +108,14 @@
 ##
 ## User privilege specification
 ##
-root ALL=(ALL) ALL
+#root ALL=(ALL) ALL
+
+##
+## allow this user to run sudo -s as root (no parameters in bash allowed)
+## users can also use 'sudo -i' to get a more real root env.
+##
+ADMINS ALL=(root) EXEC: /bin/bash ""
+ADMINS ALL=(root) sudoedit
 
 ## Uncomment to allow members of group wheel to execute any command
 # %wheel ALL=(ALL) ALL
